src/useSubscription/useSubscriptionDevtools.ts

```diff
@@ -0,0 +1,243 @@
+  refresh()
+}
+
+type UpdateChannelEventTypes = 'Loading' | 'Error' | 'Executed' | 'Response' | 'Refresh'
+type UpdateChannelEvent = {
+  [K in UpdateChannelEventTypes]: CreateSubscriptionDevtoolsTimelineEvent<K, EventTypeToDataMap[K]>
+}[UpdateChannelEventTypes]
+export function updateChannel(channel: Channel, event: UpdateChannelEvent): void {
+  if (!initialized()) { return }
+
+  channelNodes.set(channel.signature, { node: mapChannelToInspectorNode(channel), channel })
+  addTimelineEvent(event)
+  refresh()
+}
+
+export function removeChannel(channel: Channel): void {
+  if (!initialized()) { return }
+  addTimelineEvent({title: `${channel.actionName} · Channel removed`, data: {channel, action: channel.actionName}, groupId: channel.signature})
+
+  channelNodes.delete(channel.signature)
+  refresh()
+}
+
+export function registerChannelSubscription(channel: Channel, subscriptionId: number): void {
+  if (!initialized()) { return }
+  addTimelineEvent({title: `${channel.actionName} · Subscription created`, data: {channel, action: channel.actionName, subscriptionId}, groupId: channel.signature})
+
+  const channelSubscriptions = subscribedComponents.get(channel.signature) ?? new Map()
+  const vm = getCurrentInstance()
+  channelSubscriptions.set(subscriptionId, vm)
+  subscribedComponents.set(channel.signature, channelSubscriptions)
+
+  refresh()
+}
+
+export function removeChannelSubscription(channel: Channel, subscriptionId: number): void {
+  if (!initialized()) { return }
+  addTimelineEvent({title: `${channel.actionName} · Subscription removed`, data: {channel, action: channel.actionName, subscriptionId}, groupId: channel.signature})
+
+  const channelSubscriptions = subscribedComponents.get(channel.signature)
+  if (!channelSubscriptions) { return }
+  channelSubscriptions.delete(subscriptionId)
+  refresh()
+}
+
+type SubscriptionsInspectorState = CustomInspectorState & {
+  State: CustomInspectorState[keyof CustomInspectorState],
+  "Subscribed Components": CustomInspectorState[keyof CustomInspectorState],
+}
+
+function getCustomInspectorState(nodeId: string): SubscriptionsInspectorState {
+  const {channel} = channelNodes.get(nodeId as Channel['signature']) ?? {}
+  if (!channel) {
+    return {"Error": [{ key: 'message', value: 'Channel not found.'}], "State": [], "Subscribed Components": []}
+  }
+  const subscriptions = subscribedComponents.get(channel.signature) ?? new Map<number, ComponentInternalInstance | null>()
+  
+  return {
+    "State": [
+      {
+        key: 'channel',
+        value: channel,
+      }
+    ],
+    "Subscribed Components": [...subscriptions.entries()].map(([id, vm]) => (
+      {
+        key: String(id),
+        value: getComponentName(vm),
+      })
+    ),
+  }
+}
+
+function getComponentName(vm: ComponentInternalInstance | null): string {
+  // @ts-ignore __name is not in the types but works.
+  return vm?.type.__name as string ?? vm?.type?.name ?? '🤷🏻‍♂️ Unknown component'
```

Comment:
there are some cases where I haven't been able to get the actual component name... I think this might be due to some package linking issues... for example, components in ui-library are built into a prod dist and lose their debug info. or something else entirely i haven't been able to track down. 

There is commented code further up which pulls the component names from the vue devtools plugin api - it also doesn't seem to capture all names. Not sure if it's better/worse than this but I'm imagining it'd be slower than just this.